### PR TITLE
Fail when no assertions

### PR DESCRIPTION
**Pull request by HDegroote**
Currently a test can pass if it does not contain any assertions, as can an entire test run. I think that always indicates a bug in the tests.

The behaviour of this PR is:
- If a test contains 0 assertions, that test fails
- <strike>If an entire run contains 0 assertions, an error is thrown</strike>

<strike>Note: I couldn't find a clean way in the test-run lifetime to just make the run fail if there were 0 assertions, as this is known only when the `_onend` handler runs, so after the run finished already. That's why it explicitly throws an error there. (The main edge case is a run without any tests)</strike>

### CODE REVIEW

**Comment by HDegroote** [comment]
Modified the PR to take todo's into account: it now no longer throws an error when there were 0 assertions. Instead, it throws an error when there were 0 tests.
The other part of this PR  should then detect all other cases, because it detects where there are tests but they have 0 assertions

**Comment by davidmarkclements** [comment]
@HDegroote why not just make the default plan 1 instead of 0

**Comment by HDegroote** [comment]
@davidmarkclements I don't think that works, because brittle differentiates between tests with a plan and tests without a plan (`index.js` is full of `if (this._hasPlan)` guards). Using a default plan of 1 would force all tests to adopt a plan. Not sure that's what you had in mind though, so I'm happy to be corrected on this

**Comment by davidmarkclements** [comment]
@HDegroote ah yeah of course, it limits to 1

File: index.js [hunk 1]
```diff
@@ -675,8 +676,14 @@ class Test {
     const ok = (this.fails === 0)
 
     if (this.isMain && !err) {
+      const testPassed = ok && this.assertions > 0
+      const explanation = this.assertions > 0
+        ? null
+        : 'No assertions were tested'
```
Comment (davidmarkclements) [inline index.js:682]:
```suggestion
        : 'No assertions were called'
```
